src/rust/catpowder/win/runtime.rs

```diff
@@ -113,8 +115,9 @@ impl NetworkRuntime for SharedCatpowderRuntime {
             xdp_rs::_XSK_NOTIFY_FLAGS_XSK_NOTIFY_FLAG_POKE_TX | xdp_rs::_XSK_NOTIFY_FLAGS_XSK_NOTIFY_FLAG_WAIT_TX;
 
         if let Err(e) = self.0.borrow_mut().notify_socket(flags, u32::MAX, &mut outflags) {
-            warn!("failed to notify socket: {:?}", e);
-            return;
+            let cause = format!("failed to notify socket: {:?}", e);
+            warn!("{}", cause);
+            return Err(Fail::new(libc::EINVAL, &cause));
```

Comment:
We failed because due to a runtime error. In my opinion, `EAGAIN` would better reflect this condition.